new-log-viewer/src/services/LogFileManager.ts

```
@@ -230,6 +249,78 @@ class LogFileManager {
+    #searchChunk (queryId: number, beginSearchIdx: number, searchRegex: RegExp): void {
+        if (queryId !== this.#queryId) {
+            return;
+        }
+
+        const endSearchIdx = Math.min(beginSearchIdx + SEARCH_CHUNK_SIZE, this.#numEvents);
+        const results: ChunkResults = {};
+
+        for (let eventIdx = beginSearchIdx; eventIdx < endSearchIdx; eventIdx++) {
+            const contentString = this.#decoder.decode(eventIdx, eventIdx + 1)?.[0]?.[0] || "";
+            const match = contentString.match(searchRegex);
+            if (match && match.index) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)